client/src/components/profilemanager/active-value-tracker.ts

```diff
@@ -0,0 +1,52 @@
+import { Event, EventEmitter } from "vscode";
+import { sleep } from "../../sleep";
+
+export interface ActiveValueTracker<T> {
+    readonly activeChanged: Event<T>;
+    setActive(switchedTo: T): void;
+    activeAsync(): Promise<T>;
+    active(): T | null;
+}
+
+export function create<T>(getActiveValue: () => Promise<T>, pollIntervalMS: number) {
+    return new ActiveValueTrackerImpl<T>(getActiveValue, pollIntervalMS);
+}
+
+class ActiveValueTrackerImpl<T> implements ActiveValueTracker<T> {
+    private activeValue: T | null = null;
+    private readonly activeChangedEmitter: EventEmitter<T> = new EventEmitter<T>();
+
+    constructor(private readonly getActiveValue: () => Promise<T>, private readonly pollIntervalMS: number) {
+        this.pollActive();
+    }
+
+    public get activeChanged(): Event<T> {
+        return this.activeChangedEmitter.event;
+    }
+
+    public setActive(switchedTo: T): void {
+        if (switchedTo !== this.activeValue) {
+            this.activeValue = switchedTo;
+            this.activeChangedEmitter.fire(this.activeValue);
+        }
+    }
+
+    public async activeAsync(): Promise<T> {
+        const value = await this.getActiveValue();
+        this.setActive(value);
+        return value;
+    }
+
+    public active(): T | null {
+        return this.activeValue;
+    }
+
+    private async pollActive(): Promise<never> {
+        // eslint-disable-next-line no-constant-condition
+        while (true) {

```

Comment:
We are polling in case a user were to update the configuration file by hand.
Can you clarify what you mean by "Event"? Did you mean when a user executes a command?

Reply:
I have migrated away from the active-value-tracker logic to utilizing fs.watch. This allows instant feedback to the status bar when the user updates the config file. I am now closing session when a user has been updated the profile config by hand. This will most likely go away if we migrate to using a credentials.json in the future.

Reply:
I pushed a new commit to change fs.watch to chokidar, fs.watch proved to be a bit unstable: https://nodejs.org/docs/latest/api/fs.html#caveats

I have also migrated away from the commands closing session, and am now allowing our watcher to close the session for us. This way, the user will have an identical experience whether they update the profile using the status bar or manually with the file.